### DISCUSSION TITLE
help for write register 21

### DISCUSSION THREAD

**Discussion by khriss75** [celsworth/lxp-bridge#230] [Q&A]
I'd like to set in standby mode my inverter (SNA5000).
Using MQTT Explorer I try to send JSON payload with this command:
lxp/BA1xxxxxxx/hold/21/bits
{
  "eps_en": "ON",
  "ovf_load_derate_en": "OFF",
  "drms_en": "OFF",
  "lvrt_en": "OFF",
  "anti_island_en": "OFF",
  "neutral_detect_en": "OFF",
  "grid_on_power_ss_en": "OFF",
  "ac_charge_en": "OFF",
  "sw_seamless_en": "ON",
  "set_to_standby": "ON",
  "forced_discharge_en": "OFF",
  "charge_priority_en": "OFF",
  "iso_en": "OFF",
  "gfci_en": "OFF",
  "dci_en": "OFF",
  "feed_in_grid_en": "OFF"
}
but nothing change.
Reading on mqtt explorer, I can see register 21 value is 769 when inverter is ON (no standby), and value 257 when inverter is in standby mode (off mode). Sending this value again don't change the situation.

Can someone help me in order to understand who can I change the my inverter status? (in my case set "set_to_standby": "ON/OFF")

Thanks in advance

Cris

EDIT:  I already can to write/modify some setting into "standard" register eg.:  sending this "lxp/cmd/BA1xxxxxx/set/hold/125" I can change the % cut off battery discharge.

**Comment by celsworth** [comment]
Hello,

It's not possible to write the JSON back to lxp-bridge at the moment, but this makes a nice feature request actually, I might see if I can implement that.

I think writing to register 21 as you were trying to do should work. So this:

```
mosquitto_pub -t lxp/cmd/BA1xxxxxx/set/hold/21 -m 257
```

Should set the inverter to standby. Note that you're actually setting all the bits with this (so if any other bits have changed, they'll be reset to whatever 257 means), so its not ideal, but I think it should work. If it doesn't, I wonder if your inverter just isn't allowing remote control of that bit for some reason. What appears in the logs when you try? The inverter should send a reply with the new contents of the register so we can see if it worked (or maybe it returns an error)

**Reply by khriss75** [reply]
Yeeeeessssss! Thank you so much Chris!

My inverter is a LuxPower SNA5000

Sending directly 
`lxp/cmd/BA1xxx/set/hold/21` with payload 257, set the inverter in standby mode (very low power consumption from grid if battery low and no sun production). 
As the same, sending payload 769 the inverter turn operative.
If could help someone, this is the functions bit changing:
` payload 769 (switch inverter to ON/operating):
"eps_en": "ON",
  "ovf_load_derate_en": "OFF",
  "drms_en": "OFF",
  "lvrt_en": "OFF",
  "anti_island_en": "OFF",
  "neutral_detect_en": "OFF",
  "grid_on_power_ss_en": "OFF",
  "ac_charge_en": "OFF",
  "sw_seamless_en": "ON",
  "set_to_standby": "ON",    <-------
  "forced_discharge_en": "OFF",
  "charge_priority_en": "OFF",
  "iso_en": "OFF",
  "gfci_en": "OFF",
  "dci_en": "OFF",
  "feed_in_grid_en": "OFF"`

` payload 257 (switch inverter to standby/not operating but able to send and receive data):
"eps_en": "ON",
  "ovf_load_derate_en": "OFF",
  "drms_en": "OFF",
  "lvrt_en": "OFF",
  "anti_island_en": "OFF",
  "neutral_detect_en": "OFF",
  "grid_on_power_ss_en": "OFF",
  "ac_charge_en": "OFF",
  "sw_seamless_en": "ON",
  "set_to_standby": "OFF",   <-------
  "forced_discharge_en": "OFF",
  "charge_priority_en": "OFF",
  "iso_en": "OFF",
  "gfci_en": "OFF",
  "dci_en": "OFF",
  "feed_in_grid_en": "OFF"`

Just to better understand, I'm running lxp-bridge in a docker-container on windows10, data are update every 2-3 seconds, I remember that some time ago update was every 3-5 minutes... I'm lucky or have you update something?

Thanks again

Cris (without "h"  :-) )

**Reply by XanhLa96** [reply]
hi, how you can update data every 2-3 second? 
And now it work?
